src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,39 @@
 package core.basesyntax.service;
 
-import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private final StorageDaoImpl storage;
+
+    public RegistrationServiceImpl(StorageDaoImpl storage) {
+        this.storage = storage;
+    }
 
     @Override
     public User register(User user) {
+        if (user == null || user.getLogin() == null || user.getPassword() == null) {
+            throw new RegistrationException("User, login, or password cannot be null.");
```

Comment:
Don't create one big if with all conditions. Better to split one big if into several small ones to provide more informative exception messages for each case.